### PR TITLE
add more CI, instrumented runs

### PR DESCRIPTION
**Pull request by vondele**
should presumably not be merged unless CI passes on this PR.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -107,3 +107,10 @@ script:
   #
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make clean && make -j2 ARCH=x86-64-modern sanitize=undefined optimize=no debug=yes build > /dev/null && ../tests/instrumented.sh --sanitizer-undefined; fi
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make clean && make -j2 ARCH=x86-64-modern sanitize=thread    optimize=no debug=yes build > /dev/null && ../tests/instrumented.sh --sanitizer-thread; fi
+
+  #
+  # NNUE testing / TODO should work with debug=yes as well
+  #
+  - export CXXFLAGS="-O1 -fno-inline"
+  - if [ -x "$(command -v valgrind )" ]; then make clean && make -j2 ARCH=x86-64-modern debug=no optimize=no learn > /dev/null && ../tests/instrumented_learn.sh --valgrind; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make clean && make -j2 ARCH=x86-64-modern sanitize=undefined optimize=no debug=no build > /dev/null && ../tests/instrumented_learn.sh --sanitizer-undefined; fi
```
Comment (nodchip) [inline .travis.yml:116]: @vondele Could you please take a look at this line?  I think that the build target should be `learn`.

**Review by nodchip — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (vondele) [reply .travis.yml:116]: correct!

**Comment by nodchip** [comment]
`make -j2 ARCH=x86-64-modern debug=no optimize=no learn` failed for some reason in CI.  It worked with gcc 10.2.0 on MSYS2. 
 Could you please take a look at it?

**Comment by vondele** [comment]
I have rebased on current master, if it still fails I'll have a more detailed look tonight, it doesn't fail for me locally. 

Actually, I see, needs at least the additional LDFLAGS="-lstdc++fs" on this CI VM.

**Comment by nodchip** [comment]
Thank you for sending and fixing this pull request.  I confirmed that this pull request passed the CI test.  I merged it.

**Pull request merged**